devices/Bmm150/Bmm150I2cBase.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Magnetometer
+{
+    /// <summary>
+    /// Abstract class for Bmm150 the I2C interface. This sensor can be found as a sub
+    /// I2C sensor like in the MPU9250. The access is done thru another I2C device and the
+    /// core I2C primitive are different. Use those 3 primitive to define the access to read
+    /// and write bytes to the Bmm150
+    /// </summary>
+    public abstract class Bmm150I2cBase
```

Comment:
This is not needed. The AK8963 has this because it was used in the MPU9250 as a hidden sensor that was not exposed and needed to be access in a special way. But in the case of the BMM150, I don't think it's like this. Maybe check in the BMI160 documentation to see how the magnetometer is connected. And if it's the same mechanism, then keep this otherwise, just simplify!

Reply:
As discussed, we keep this interface. 
Refer to the BMI160 [specsheet](https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmi160-ds000.pdf) (page 65) for more details.